labs/src/main/java/org/gbif/pipelines/labs/performance/CompressionTestBuilder.java

```diff
@@ -0,0 +1,96 @@
+package org.gbif.pipelines.labs.performance;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+
+import org.apache.avro.file.CodecFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Compression Performance Test Builder, which takes inputs to configure tests and execute it.
+ */
+public class CompressionTestBuilder {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CompressionTestBuilder.class);
+
+  private final List<Path> datasets;
+  private List<Integer> syncIntervals;
+  private int repeatTests = 1;
+  private List<CodecFactory> codecFactories;
+
+  /**
+   * Path of datasets.
+   */
+  public static CompressionTestBuilder forAll(Path[] datasets) {
+    return new CompressionTestBuilder(Arrays.asList(datasets));
+  }
+
+  private CompressionTestBuilder(List<Path> datasets) {
+    this.datasets = datasets;
+  }
+
+  /**
+   * list of syncIntervals to try all pair of datasets,codec combinations.
+   */
+  public CompressionTestBuilder forEach(Integer[] syncIntervals) {
+    this.syncIntervals = Arrays.asList(syncIntervals);
+    return this;
+  }
+
+  /**
+   * Repeat tests for every unique combination of datasets , codecfactory and syncInterval tuple, to get average results.
+   */
+  public CompressionTestBuilder times(int repeatTests) {
+    this.repeatTests = repeatTests;
+    return this;
+  }
+
+  /**
+   * list of codecs to apply for the pairs of dataset.
+   */
+  public CompressionTestBuilder withEach(CodecFactory[] codecFactories) {
+    this.codecFactories = Arrays.asList(codecFactories);
+    return this;
+  }
+
+  /**
+   * Execute compression function and get compression results.
+   */
+  public List<CompressionResult> performTestUsing(Function<CompressionRequest, CompressionResult> datasetFunction) {
+    List<CompressionResult> results = new ArrayList<>();
```

Comment:
you can change all these to something like: 
List<CompressionResult> results = new ArrayList<>();
    datasets.forEach(dataset ->
      codecFactories.forEach(codecFactory ->
        syncIntervals.forEach( syncInterval ->
          results.add(datasetFunction.apply(new CompressionRequest(dataset,
                                                                   syncInterval,
                                                                   repeatTests,
                                                                   codecFactory)))
        )
      )